community-version.py

```diff
@@ -7,32 +7,56 @@
 import sys
 import argparse
 import os
-from PIL import Image, ImageDraw, ImageFont
+from PIL import Image, ImageDraw, ImageFont, ImageEnhance
 import cmd
 from example.make_art import convert_image_to_ascii
 import requests
 from io import BytesIO
 
-
 # changing ascii-art to image
-text_file = "./custom_text.txt"
-def art_to_image(text_file):
+text_file = "./test/ztm-logo.txt"
```

Comment:
This looks like code you changed for testing purposes?
There is no test/ztm-logo.txt file.

Reply:
I have created a **test** folder to organize all types of input and output files. It seems that this **test** folder should not be part of the git repo, so I will make the necessary adjustments.

Could you please confirm the steps I should take next?:
1.Execute '**git pull upstream master**'
2.Modify **text_file** and commit the changes

Additionally I have another question. I noticed that small portion of my code is changed to 
**image = Image.new("RGBA", (0, 0))
draw = ImageDraw.Draw(image)
(_, _, right, bottom) = draw.multiline_textbbox((0, 0), ascii_text) # get width and height in px**

when I tried this out, I obtained the following output:

![output_ztm-logo](https://github.com/zero-to-mastery/ascii-art/assets/132012265/416684eb-e483-413a-9f35-6f3efdad43ae)

Could you kindly clarify?

Reply:
@kennithnichol 
-> Made the required changes.
-> Adjusted the aspect ratio of the ASCII art by using two characters per pixel. This makes it more visually similar to the original image without stretching.
-> Although ASCII_CHARS = list(ascii_printable) is used, only the first few characters of ASCII_CHARS were actually considered(['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a']), due to the [min((pixel_value // range_width), len(ASCII_CHARS) - 1)]. This limited the clarity of the ASCII art. Therefore, I switched to ['#', '?', '%', '.', 'S', '+', '.', '*', ':', ',', '@'].